### ISSUE TITLE
🐛 pip install of optional dependencies is broken

### ISSUE THREAD

**Issue by malmans2** [Cloud-Drift/clouddrift#448]
**Describe the bug**
The installation of optional dependencies using pip seems to be broken as described in the documentation. 

**To Reproduce**
```
mamba create -n clouddrift -c conda-forge python=3.11 pip
mamba activate clouddrift
pip install clouddrift
pip install matplotlib-base cartopy
```

**Desktop (please complete the following information):**
 - Operating System: Darwin MacBook-Pro-3.local 22.6.0 Darwin Kernel Version 22.6.0: Mon Apr 22 20:54:28 PDT 2024; root:xnu-8796.141.3.705.2~1/RELEASE_X86_64 x86_64
 - clouddrift==0.35.0

**Comment by malmans2** [comment]
Oops, I forgot the traceback.
Here it is:
```
ERROR: Could not find a version that satisfies the requirement matplotlib-base (from versions: none)
ERROR: No matching distribution found for matplotlib-base
```

**Comment by philippemiron** [comment]
Could reproduce but not sure why it's not working with pip.

This fails:
```
python3 -m venv .venv
source .venv/bin/activate

pip install clouddrift
pip install matplotlib-base cartopy
```
but installing the complete `matplotlib` instead of `matplotlib.base` works.

Using `conda-forge` completely also works.
```
conda env create --name clouddrift-test
conda activate clouddrift-test
conda install clouddrift
conda install matplotlib-base cartopy
```

**Comment by philippemiron** [comment]
Looks like something is wrong with their package because this doesn't work:
```
conda create -n clouddrift-test -c conda-forge python=3.11 pip
conda activate clouddrift-test
pip install matplotlib-base
```

**Comment by malmans2** [comment]
If you are happy with showing just the conda instructions, I'm fine with it.
Generally, I would recommend instructing users to follow the same approach used in the CI.

**Comment by philippemiron** [comment]
I think the idea was to allow for using `pip` directly and also support `conda` since a lot of people use it in Physical Oceanography.

Indeed CI uses conda.
```
      - uses: mamba-org/setup-micromamba@v1
        with:
          environment-file: environment.yml
          environment-name: clouddrift
          create-args: >-
            python=${{ matrix.python-version }}
            pytest
            coverage
            matplotlib-base
            cartopy
```
so we didn't see this error with `pip`.

**Comment by kevinsantana11** [comment]
fixed by #451
